### DISCUSSION TITLE
Is it possible to have a module with many subpackages?

### DISCUSSION THREAD

**Discussion by egor-ponomarev** [spring-projects/spring-modulith#179] [Q&A] [answered]
Hello all! 
I'd like to use spring-modulith to have a project with several modules, each of them has a package 'api' which should be available only for some of other modules and which can contain lots of subpackages which also should be available for all modules that have access on the package 'api'. I tried to implement it using @NamedInterface but looks like it is violates the rules. 
Would it be possible to have the described structure?

Here is what structure I'd like to have (rest, service, repository are the modules) 
![example-structure](https://user-images.githubusercontent.com/19322181/230354361-0f617550-5392-4daf-bc31-07c47b81a829.png)

**Comment by odrotbohm** [comment, marked as answer]
Your package structure implies that you implement layering, not modules. If you find the most relationships between your type to cross package boundaries, that is a smell. [Here's](https://www.youtube.com/watch?v=SjSjBZ7mo1g) a talk I gave recently that describes the problems with a rather detailed, technical package arrangement.
